services/repository/files/content.go

```diff
@@ -214,7 +214,9 @@ func GetContents(ctx context.Context, repo *repo_model.Repository, treePath, ref
 		if err != nil {
 			return nil, err
 		}
-		contentsResponse.SubmoduleGitURL = &submodule.URL
+		if submodule != nil {
```

Comment:
it's probably an edge case but imho would be better to also check if URL is set:
```suggestion
		if submodule != nil && len(submodule.URL) != 0 {
```

Reply:
```suggestion
		if submodule != nil && submodule.URL != "" {
```